test-apps/test-tutorial-servo-can-control/src/TutorialServo.cpp

```diff
@@ -0,0 +1,36 @@
+#include "TutorialServo.h"
+#include "mbed.h"
+
+TutorialServo::TutorialServo(PinName servoPin, float servoRangeInDegrees, float minPulsewidthInMs, float maxPulsewidthInMs): m_servoPwmOut(servoPin), m_servoRangeInDegrees(servoRangeInDegrees), m_minPulsewidthInMs(minPulsewidthInMs), m_maxPulsewidthInMs(maxPulsewidthInMs) {
+        m_servoPwmOut.period_ms(20);
+}
+
+float TutorialServo::getServoRangeInDegrees( ) const{
+    return m_servoRangeInDegrees;
+}
+
+float TutorialServo::getMinPulseWidthInMs( ) const{
+    return m_minPulsewidthInMs;
+}
+
+float TutorialServo::getMaxPulseWidthInMs( ) const{
+    return m_maxPulsewidthInMs;
+}
+
+void TutorialServo::setPositionInDegrees(const float degrees){
+    if (degrees < 0){
+        printf("Out of Range. Setting to min");
+        m_servoPwmOut.pulsewidth(m_minPulsewidthInMs);
+        return;
+    }
+    else if(degrees > m_servoRangeInDegrees){
+        printf("Out of Range. Setting to max");
+        m_servoPwmOut.pulsewidth(m_maxPulsewidthInMs);
+        return;
+    }
+    float slope = (m_maxPulsewidthInMs - m_minPulsewidthInMs)/m_servoRangeInDegrees;
+    float pulse = m_minPulsewidthInMs + degrees*slope;
+    printf("Setting to %d",degrees);
+    m_servoPwmOut.pulsewidth(pulse);
```

Comment:
The pulse you calculated is in ms, but pulsewidth() accepts a value in seconds (https://os.mbed.com/docs/mbed-os/v6.11/mbed-os-api-doxy/classmbed_1_1_pwm_out.html#add3053f7f15f458c214b3f9c70a02f16), so you'll need to convert from ms to s first.